R/orcid_citations.R

```diff
@@ -219,9 +219,18 @@ do_all <- function(m, orcid, put_code, cr_format, cr_style, cr_locale, ...) {
   without_doi_citations <- lapply(without_doi, function(z) {
     zzz <- z$`external-ids.external-id`[[1]] %||% z$`work.external-ids.external-id`[[1]]
     pc <- z$`put-code` %||% z$`work.put-code`
-    list(put = pc, ids = zzz$`external-id-value`, 
-      type = zzz$`external-id-type`, format = cr_format,
-      citation = cite_put(orcid, pc))
+
+    ##try to extract BibTeX, if this fails use the conventional way
+    citation <- try(
+      extract_BibTeX(cite_put(orcid, pc, ctype = "application/vnd.orcid+json; qs=4")))
+
+    if(class(citation) == "try-error")
```

Comment:
use `inherits(citaiton, "try-error")` instead of `class(citation) == "try-error"`

Reply:
Done